lib/utils.pm

```diff
@@ -2371,16 +2372,30 @@ This functions checks if ca-certificates-suse is installed and if it is not it a
 
 sub ensure_ca_certificates_suse_installed {
     return unless is_sle || is_sle_micro;
+    my $ret = 0;
     if (script_run('rpm -qi ca-certificates-suse') == 1) {
         my $host_version = get_var("HOST_VERSION") ? 'HOST_VERSION' : 'VERSION';
         my $distversion = get_required_var($host_version) =~ s/-SP/_SP/r;    # 15 -> 15, 15-SP1 -> 15_SP1
-        zypper_call("ar --refresh http://download.suse.de/ibs/SUSE:/CA/SLE_$distversion/SUSE:CA.repo");
-        if (is_sle_micro) {
-            transactional::trup_call('--continue pkg install ca-certificates-suse');
+
+        # exit codes defined will not stop on error.
+        my $out = zypper_call("ar --refresh http://download.suse.de/ibs/SUSE:/CA/SLE_$distversion/SUSE:CA.repo", retry => 3, exitcode => [0, 1, 2, 3, 4]);
+        if ($out == 0) {
+            if (is_sle_micro) {
+                $ret = transactional::trup_call('--continue pkg install ca-certificates-suse');
+            } else {
+                $ret = zypper_call("in ca-certificates-suse");
+            }
         } else {
-            zypper_call("in ca-certificates-suse");
+            # Get SUSE SSL certificates from internal CA
+            script_retry('curl -k https://ca.suse.de/certificates/ca/SUSE_Trust_Root.crt -o /etc/pki/trust/anchors/SUSE_Trust_Root.crt', timeout => 100, delay => 30, retry => 5);
+            if (is_sle_micro) {
+                $ret = transactional::trup_call('--continue run update-ca-certificates -v');
+            } else {
+                $ret = assert_script_run 'update-ca-certificates -v';
+            }
         }
     }
+    return $ret;
```

Comment:
I am not sure whether returning the caller a int makes it any better. By the name the function should either succeed or die.

Reply:
Ok, but let we consider some new situation raised, due to  [IBS CA](http://download.suse.de/ibs/SUSE:/CA/), where NO sle-micro present: because of that, I introduced the `out` : `true` for SLE(S)/OpenSUSE OR `fail` (SLE-Micro or else) and result became a 4 way output.
IMHO the decision hould be passed from caller by  a more option  `continue_on_fail` or die: 
wdyt ?

Reply:
until 6.0, sle-micro versions were mapped to sle, which is the previous code reflecting. The question is how we should continue over here and that is a question on PjM rather than any QA engineer can answer.

Reply:
Please check last commit.

Reply:
I think I have found the cert folder, or at least it might be worth to be tested `/SLE-Factory/` in case of slem 6.0+

Reply:
I already noted it, but I had no info on if-and-when it can be used, in place of which case.

Reply:
> I already noted it, but I had no info on if-and-when it can be used, in place of which case.

First of all I would try to test it, then you can come to the project channel with direct questions. We are not building the assets so we cannot know :)